python/hail/methods/statgen.py

```diff
@@ -204,6 +204,179 @@ def linreg(dataset, ys, x, covariates=[], root='linreg', block_size=16):
 
     return cleanup(MatrixTable(jm))
 
+@handle_py4j
+@typecheck(dataset=MatrixTable,
+           key_expr=expr_any,
+           weight_expr=expr_numeric,
+           y=oneof(expr_numeric, expr_bool),
+           x=expr_numeric,
+           covariates=listof(oneof(expr_numeric, expr_bool)),
+           logistic=bool,
+           max_size=integral,
+           accuracy=numeric,
+           iterations=integral)
+def skat(dataset, key_expr, weight_expr, y, x, covariates=[], logistic=False,
+         max_size=46340, accuracy=1e-6, iterations=10000):
+    r"""Test each keyed group of variants for association by linear or logistic
+    SKAT test.
+
+    Examples
+    --------
+
+    Test each gene for association using the linear sequence kernel association
+    test:
+
+    >>> ds = hc.read('data/example_burden.vds')
+    >>> skat_table = methods.skat(ds,
+    ...                           key_expr=ds.gene,
+    ...                           weight_expr=ds.weight,
+    ...                           y=ds.burden.pheno,
+    ...                           x=ds.GT.num_alt_alleles(),
+    ...                           covariates=[ds.burden.cov1, ds.burden.cov2])
+
+    .. caution::
+
+       By default, the Davies algorithm iterates up to 10k times until an
+       accuracy of 1e-6 is achieved. Hence a reported p-value of zero with no
+       issues may truly be as large as 1e-6. The accuracy and maximum number of
+       iterations may be controlled by the corresponding function parameters.
+       In general, higher accuracy requires more iterations.
+
+    .. caution::
+
+       To process a group with :math:`m` variants, several copies of an
+       :math:`m \times m` matrix of doubles must fit in worker memory. Groups
+       with tens of thousands of variants may exhaust worker memory causing the
+       entire job to fail. In this case, use the `max_size` parameter to skip
+       groups larger than `max_size`.
+
+    Notes
+    -----
+
+    This method provides a scalable implementation of the score-based
+    variance-component test originally described in
+    `Rare-Variant Association Testing for Sequencing Data with the Sequence Kernel Association Test
+    <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3135811/>`__.
+
+    The test is run on complete samples (i.e., phenotype and all covariates
+    non-missing). For each variant, missing input values are imputed as the
+    mean of all non-missing input values.
+
+    Variant weights must be non-negative. Variants with missing weights are
+    ignored. In the R package ``skat``, default weights are given by evaluating
+    the Beta(1, 25) density at the minor allele frequency. To replicate these
+    weights in Hail using alternate allele frequencies stored in a row-indexd
+    field `AF`, one can use the expression:
+
+    ``functions.dbeta(AF[0].min(AF[1]), 1.0, 25.0) ** 2``
```

Comment:
I think this would be better as a code block `>>>` (looks nicer and will be tested to guard against future interface changes). Feel free to push back if it's too difficult to make the example work or it's not as clear with the extra text referring to `ds`.